### PR TITLE
[IOTDB-665] validate physical plan before fowarding

### PR DESCRIPTION
**Pull request by Ring-k**
This pull request implements physical plan validation before fowarding. The client connects and sends requests to a node in the cluster, which should response some messages if the statement is invalid.

The validation is implemented in `MetaGroupMember` class, `validatePlan()` method.

```
Currently implemented:
    - SetStorageGroupPlan: 
        + path should not exist
    - InsertPlan:
        + path should exist
        + the value can be parsed into the data type of the path
```

### CODE REVIEW

File: cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java [hunk 1]
```diff
@@ -1397,6 +1401,91 @@ public TSStatus executeNonQuery(PhysicalPlan plan) {
...
+    }
+    // TODO add more pre-validations
+    return result;
+  }
+
+  private TSStatus validateSetStorageGroupPlan(SetStorageGroupPlan plan) {
+    String path = plan.getPath().getFullPath();
+    if (getAllStorageGroupNames().contains(path)) {
+      return StatusUtils.PATH_ALREADY_EXIST_ERROR;
+    }
```
Comment (jt2594838) [inline cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java:1420]: I think you should also check the legalness of the path. It should start with "root' but cannot be "root".

File: cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java [hunk 2]
```diff
@@ -1397,6 +1401,91 @@ public TSStatus executeNonQuery(PhysicalPlan plan) {
...
+  private TSStatus validateInsertPlan(InsertPlan plan) {
+    List<Path> paths = (plan).getPaths();
+    String[] values = (plan).getValues();
+    List<String> pathStrings = new ArrayList<>();
+    for (Path path : paths) {
+      pathStrings.add(path.getFullPath());
+    }
+    List<MeasurementSchema> measurementSchemas;
+    try {
+      measurementSchemas = pullTimeSeriesSchemas(pathStrings);
```
Comment (jt2594838) [inline cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java:1433]: Check the schema cache in MManager using MManager.getSeriesType() first, if it cannot be found then add them to the list and pull them.

File: cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java [hunk 3]
```diff
@@ -1397,6 +1401,91 @@ public TSStatus executeNonQuery(PhysicalPlan plan) {
...
+      case INT32:
+        Integer.parseInt(value);
+        break;
+      case INT64:
+        Long.parseLong(value);
+        break;
+      case BOOLEAN:
+        if (!(value.equals("true") || value.equals("TRUE")
+            || value.equals("false") || value.equals("FALSE")
+            || value.equals("0") || value.equals("1"))) {
```
Comment (jt2594838) [inline cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java:1473]: We have defined these strings in SQLConstants, better to use them. And maybe you can use `equalsIgnoreCase` to increase robustness.

File: cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java (hunk 1 above)
Comment (Ring-k) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java:1420]: That's right. Thanks for your reminding.

File: cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java (hunk 2 above)
Comment (Ring-k) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java:1450]: The schema cache can help to accelerate the validations. Thanks for your suggestions.

File: cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java (hunk 3 above)
Comment (Ring-k) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java]: Oh, I didn't notice that. That's convenient. Thanks for your suggestion :)

**Pull request merged**